abnf/odata-abnf-construction-rules.txt

```diff
@@ -910,11 +910,11 @@ primitiveLiteral = nullValue                  ; plain values up to int64Value
                  / geometryPolygon
 
 ; in Atom and JSON message bodies and CSDL DefaultValue attributes
-primitiveValue = booleanValue
-               / guidValue
+primitiveValue = boolean

```

Comment:
`primitiveValue` does not include string values, although they can appear in DefaultValue attributes (unquoted). They will also occur as key values in context URLs (quoted, but not percent-encoded).

Do we need additional rules `stringValue` and `stringValueQuoted`? (Unless we delete the `primitiveValue` rule completely: it is not used anywhere else in our ABNF.)

Reply:
This rule is incomplete, see comment at the end of the rule:
https://github.com/oasis-tcs/odata-abnf/blob/9d42aa71b7644f015ca7efcf59c260aedc1a3052/abnf/odata-abnf-construction-rules.txt#L936-L938

Reply:
I'm tempted to completely remove this rule

Reply:
If we omit `stringValue`, we could also omit `int16Value`, because it does not (yet) appear elsewhere and is also covered by the JSON remark. Both `int16Value` and `stringValueQuoted` will appear in the new `context` rule (#106), so why not already define both here?

Reply:
> I'm tempted to completely remove this rule

And then define `int16Value`, ..., and `stringValue` (quoted) only when they are needed for the new `context` rule?

Reply:
Cross-checked with specifications:
- only `DefaultValue` text for CSDL XML references rule `primitiveValue` for values of primitive types other than `Edm.String`, so the rule is correct, removed the outdated trailing comment.

Reply:
```
<Term Name="..." Type="Edm.String" DefaultValue="ARGH!@XXX" />
```
is valid CSDL XML, but `ARGH!@XXX` does not match the `primitiveValue` rule.